### PR TITLE
experimental gtfobin shell breakout detections

### PR DESCRIPTION
**Pull request by Murphy0801**
### Summary of the Pull Request

Recently we developed and tested 11 new rules which detect GTFOBINs shell breakouts, and we would like to contribute back to SigmaHQ repo, as we didn't find existing ones. The rules we are contributed are tested in our environment for a period of time for potential false positives. Rules are also validated by MSV (Mandiant Security validation) tests in our setup for true detections.

### Changelog

new: Capsh Shell Invocation - Linux
new: Inline Python Execution - Spawn Shell Via OS System Library
new: Shell Execution GCC  - Linux
new: Shell Execution via Find - Linux
new: Shell Execution via Flock - Linux
new: Shell Execution via Git - Linux
new: Shell Execution via Nice - Linux
new: Shell Execution via Rsync - Linux
new: Shell Invocation via Env Command - Linux
new: Shell Invocation Via Ssh - Linux
new: Suspicious Invocation of Shell via AWK - Linux

### Example Log Event
N/A

### Fixed Issues
N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)
- We read and applied the conversions and passed the tests.

### CODE REVIEW

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_rsync_shell.yml
Comment (frack113) [inline rules/linux/process_creation/proc_creation_lnx_gtfobin_rsync_shell.yml:23]: Can not use `sh 1>&2 0<&2` to bypass the rule ?
contains 'sh...' cover 'bash...' and 'dash...' or other *sh like zsh

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml
Comment (frack113) [inline rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml:9]:
```suggestion
date: 2024-08-05
```

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml
Comment (frack113) [inline rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml:20]: Need a log to check but I think ";" will split the commandline in 2 logs.

**Review by frack113 — COMMENTED** [review]
Hi,
I have made a FAST review from reference only as there are no Example Log Event.
The SigmaHQ repo use the [V2 specification](https://github.com/SigmaHQ/sigma-specification) now

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_git_shell.yml
Comment (Murphy0801) [reply rules/linux/process_creation/proc_creation_lnx_gtfobin_git_shell.yml:26]: Based on our MSV test for git breakout, example logs are following (partial logs but you can see the pattern):
process_image  : /usr/bin/bash                                                    
process_cmdline :  /bin/sh -c "sh -c \"exec sh 0<&1\"" "sh -c \"exec sh 0<&1\""     
process_parent_image : /usr/bin/git                                                     
process_parent_cmdline : git -p help

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_rsync_shell.yml
Comment (Murphy0801) [reply rules/linux/process_creation/proc_creation_lnx_gtfobin_rsync_shell.yml]: Good point! please review the latest commit for the detection:
    selection_cli:
        CommandLine|contains:
            - 'sh 0<&2 1>&2'
            - 'sh 1>&2 0<&2'
           
This logic will cover the mentioned bypass and *sh

File: rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml
Comment (Murphy0801) [reply rules/linux/process_creation/proc_creation_lnx_gtfobin_ssh_shell.yml:20]: We checked the logs, we can see the original log with commandline contains "ProxyCommand=;", and as well as child processes, see below example logs.

This rule is aiming to detect three SSH variants (https://gtfobins.github.io/gtfobins/ssh/#shell):

SSH Variant 1:
process_image :  /usr/bin/ssh                                                                         
process_cmdline  : ssh localhost /bin/bash --noprofile --norc   
SSH Variant 2:
process_image : /usr/bin/ssh                                                                         
process_cmdline  : ssh -o "proxycommand=;sh 0<&2 1>&2" x  
SSH Variant 3:
process_image  : /usr/bin/ssh                                                                         
process_cmdline  : ssh -o permitlocalcommand=yes -o localcommand=/bin/sh xxxx.com

**Comment by Murphy0801** [comment]
@nasbench @frack113 , following up on the PR, any other recommendations or questions regarding to the rules?

**Comment by nasbench** [comment]
> @nasbench @frack113 , following up on the PR, any other recommendations or questions regarding to the rules?

Still haven't fully worked on this yet. I'll let you know if I have any questions, gonna start working on it soon. Thanks for your patience.

**Review by nasbench — APPROVED** [review]

**Pull request merged**
